helix-core/src/syntax.rs

```diff
@@ -1808,6 +1825,7 @@ fn injection_for_match<'a>(
     }
 
     let mut include_children = false;
+    let mut include_unnamed_children = false;
```

Comment:
Instead of two booleans, I think it would be cleaner to use an enum

Reply:
I think a enum with tree members makes the most sense here :+1: